File: 75-sort-colors/75-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n=len(nums)
        low=0
        mid=0
        high = n-1
        while (mid<=high):
            if nums[mid]==0:
                (nums[low],nums[mid])=(nums[mid],nums[low])
                low+=1
                mid+=1
            elif (nums[mid]==1):
                mid+=1
            else:
                (nums[mid],nums[high])=(nums[high],nums[mid])
                high-=1
        

File: 75-sort-colors/75-sort-colors.java
class Solution {
    //Using two-pointer approach and three-pointer approach
    //Two pointer + helper pointer approach
    //if mid pointer surpasses high pointer, the array is sorted
    public void sortColors(int[] nums) {
        int len = nums.length;
        int low = 0;
        int mid = 0;
        int high = len-1;
        while(mid<=high){
            if(nums[mid]==0){
                int temp = nums[low];
                nums[low] = nums[mid];
                nums[mid] = temp;
                low +=1;
                mid +=1;
            }
            else if(nums[mid]==1)
                mid+=1;
            else{
                int temp = nums[mid];
                nums[mid] = nums[high];
                nums[high] = temp;
                high -=1;
            }
        }
        
        
    }
}